docker-compose.yml

```diff
@@ -18,7 +18,10 @@ services:
       # Uncomment if application.properties needs to be customized. Create a config folder and put customized application.properties file in it.
       # - "./config/application.properties:/usr/local/tomcat/data/config/application.properties:ro"
     environment:
-      - JAVA_OPTS=-Dspring.kafka.bootstrap-servers=kafka:9092 -Dspring.rabbitmq.host=rabbitmq -Dspring.rabbitmq.username=admin -Dspring.rabbitmq.password=admin
+      JAVA_OPTS: >-
+        -Dspring.kafka.bootstrap-servers=kafka:9092
+        -Dspring.rabbitmq.host=rabbitmq -Dspring.rabbitmq.username=admin -Dspring.rabbitmq.password=admin
+        -Dspring.security.oauth2.client.provider.keycloak.realm-uri=http://${EXTERNAL_HOSTNAME:-host.docker.internal}:8081/realms/demo-app-realm
```

Comment:
I think we shouldn't assume the port is 8081 for external keycloak server. Please make that customizable like the `EXTERNAL_HOSTNAME`.